diff --git a/src/runtime/uv/event_loop.cpp b/src/runtime/uv/event_loop.cpp
--- a/src/runtime/uv/event_loop.cpp
+++ b/src/runtime/uv/event_loop.cpp
@@ -148,3 +148,3 @@
-/* Std.Internal.UV.Loop.alive : BaseIO UInt64 */
-extern "C" LEAN_EXPORT lean_obj_res lean_uv_event_loop_alive() {
-    return io_result_mk_error("lean_uv_event_loop_alive is not supported");
+/* Std.Internal.UV.Loop.alive : BaseIO Bool */
+extern "C" LEAN_EXPORT uint8_t lean_uv_event_loop_alive() {
+    return 0;
diff --git a/src/runtime/uv/system.cpp b/src/runtime/uv/system.cpp
--- a/src/runtime/uv/system.cpp
+++ b/src/runtime/uv/system.cpp
@@ -637,1 +637,2 @@
-extern "C" LEAN_EXPORT lean_obj_res lean_uv_os_get_group() {
+extern "C" LEAN_EXPORT lean_obj_res lean_uv_os_get_group(uint64_t gid) {
+    (void)gid;
